### ISSUE TITLE
JS ternary expression not colored correctly

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/TypeScript-TmLanguage#81]
Under vscode master (0.10.9) with language mode JavaScript.

![ternary](https://cloud.githubusercontent.com/assets/2193314/12963913/0eee1674-d002-11e5-8d01-08f8609b26e1.png)

`'amd64'` has classes: token block decl expr function meta single string ts var var-single-variable js
`'i386'` has classes: token **annotation** block decl expr function meta single string ts type var var-single-variable js

`annotation` scope is causing the following theme style to be applied:

```css
.monaco-editor.vs-dark.vscode-theme-colorful-defaults-themes-dark_plus-tmTheme .token.var.annotation.type.js {
    color: rgba(78, 201, 176, 1);
}
```

**Comment by thorn0** [comment]
Still happens with vscode 0.10.10.

**Comment by SamVerschueren** [comment]
Should the ternary operator be declared on itself? I searched through the document but couldn't find a direct match to be honest.

**Comment by Tyriar** [comment]
You generally only scope something if it's useful for styling wouldn't you? I would expect the `?` and `:` to be scoped as symbols or operators?

**Comment by SamVerschueren** [comment]
Does this mean there is a false-positive match which adds the `annotation` style? Not very familiar with how this all works. Trying to get a grip on it :).

**Comment by Tyriar** [comment]
I'm not sure what annotation is supposed to me, some TypeScript syntax related to `:`? I do notice however that removing the `var` fixes the issue so this seems specific to declaring and initializing variables with ternary operators.

![image](https://cloud.githubusercontent.com/assets/2193314/15584713/9ba827a4-2330-11e6-9e3e-49cd547c5b02.png)

**Comment by zhengbli** [comment]
Fixed by #129
